skfda/ml/regression/_linear_regression.py

```diff
@@ -330,3 +400,49 @@ def _argcheck_X_y(
                 )
 
         return new_X, y, sample_weight, coef_info
+
+    def __dataframe_conversion(
+        self,
+        X: pd.DataFrame,
+    ) -> List[AcceptedDataType]:
+        """Convert DataFrames to a list with two elements.
+
+        First of all, a list with mv covariates and the second,
+        a FDataBasis object with functional data
+
+        Args:
+            X: pandas DataFrame to convert
+
+        Returns:
+            List: first of all, a list with mv covariates
+            and the second, a list of FDataBasis object with functional data
+        """
+        mv_list = []
+        fd_list = []
+        final = []
+
+        for obs in X.values.tolist():
```

Comment:
You need to iterate over the covariates directly, appending them to a list. I think you can just do
```python
return [v.values for k, v in X.items()]
```
which is both easier and more efficient.

Reply:
This way does not support columns of single numbers instead of array. We talked about allow users to use dataframes in two ways:
1) 
`
multivariate1 = [0, 2, 1, 3, 4, 2, 3]
multivariate2 = [0, 7, 7, 9, 16, 14, 5]
cov_dict = {"fd": x_fd, "mult1": multivariate1, "mult2": multivariate2}
df = pd.DataFrame(cov_dict)
`
2) 
`
multivariate = [[0, 0], [2, 7], [1, 7], [3, 9], [4, 16], [2, 14], [3, 5]]
cov_dict = {"fd": x_fd, "mult": multivariate}
df = pd.DataFrame(cov_dict)
`

Reply:
Why does it not support that?